go/obscuronode/enclave/enclave.go

```diff
@@ -221,9 +226,19 @@ func (e *enclaveImpl) SubmitRollup(rollup common2.ExtRollup) {
 }
 
 func (e *enclaveImpl) SubmitTx(tx common2.EncryptedTx) {
-	t := DecryptTx(tx)
-	e.db.StoreTx(t)
-	e.txCh <- t
+	decryptedTx := DecryptTx(tx)
+	verifySignature(decryptedTx)
+	e.db.StoreTx(decryptedTx)
+	e.txCh <- decryptedTx
+}
+
+// Checks that the L2Tx has a valid signature.
+func verifySignature(decryptedTx L2Tx) {
+	signer := types.NewLondonSigner(big.NewInt(ChainID))
+	_, err := types.Sender(signer, &decryptedTx)
+	if err != nil {
+		panic(fmt.Errorf("sig didn't validate: %w", err))
```

Comment:
I'd say we return the 'err' from this function instead of panicking.
It is expected behaviour after all that some signatures will fail

Reply:
Done